File: legacy/MoneyAI/ParentChildMatchers/EtsyOrderMatcher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CommonUtils;

namespace MoneyAI.ParentChildMatchers
{
    public class EtsyOrderMatcher : GenericOrderMatcher
    {
        public EtsyOrderMatcher(AccountInfo accountInfo)
            : base(accountInfo, @"total_shipping_cost", @"total_tax_cost", @"discount_amt")
        {
        }
    }
}

File: packages/core/src/matching/etsy-order-matcher.ts
/**
 * Etsy order-history parent-child matcher.
 *
 * Ported from C# `EtsyOrderMatcher`. A thin subclass of
 * {@link GenericOrderMatcher} that supplies Etsy-specific
 * provider attribute names for shipping, tax, and discount.
 *
 * @module
 */

import type { AccountInfo } from '../models/account-info.js';
import { GenericOrderMatcher } from './generic-order-matcher.js';

export class EtsyOrderMatcher extends GenericOrderMatcher {
    constructor(accountInfo: AccountInfo) {
        super(accountInfo, 'total_shipping_cost', 'total_tax_cost', 'discount_amt');
    }
}
